tgarchive/__init__.py

```diff
@@ -116,7 +119,34 @@ def main():
         ))
 
         try:
-            Sync(cfg, args.session, DB(args.data)).sync(args.id)
```

Comment:
This big block of logic shouldn't ideally be here where command line flags are handled. Could you move this to the `Sync` class? Maybe a new method `new_client(cfg)` that the constructor calls to either get a normal client or a takeout client based on the configuration.

Reply:
I need some guidance with this. The takeout() method returns a `_TakeoutClient` which is a context manager and its methods have to be used from within the with block, which makes sharing it across methods as an instance attribute (like the standard `TelegramClient`) of `Sync` challenging.

Reply:
`with` internally calls `__enter__()` on initialization and `__exit__()` on finish. Inside `Sync`, you could get a takeout client instance by doing something like `takeout = client.takeout(finalize=True).__enter__()`. Once the whole routine is done, calling `takeout.__exit__()` should close it.

Reply:
Thanks! I feel so stupid for trying out a few things but not this. I have made the changes as suggested and will push them soon.